### PR TITLE
Endpoints to create an auth realm and retrieve all auth realms for an account

### PR DESCRIPTION
**Pull request by vidyanambiar**
This is an initial implementation of the following endpoints:
POST /auth_realms
GET /auth_realms

### DB setup
The endpoints can be tested locally with the following DBs:
- sqlite DB (default) 
- Postgres DB (with the following env variables set)
```
                "DATABASE": "pgsql",
                "PGSQL_USER": <user>,
                "PGSQL_PASSWORD": <password>,
                "PGSQL_HOSTNAME": "127.0.0.1",
                "PGSQL_PORT": 5432,
                "PGSQL_DATABASE": <dbname>
```

### Setting X-Rh-Identity header to get an Account number
See [this doc](https://docs.google.com/document/d/1PAzJqcUXlxg7t5cX1lsPsQtBPT_95bZbyB9Iiv_ekGM/edit#) for the structure of the X-Rh-Identity header that contains the user's account number. The doc contains an example JSON as well. This JSON needs to be base64 encoded and passed as a header to the API endpoint for the account number to be parsed by the middleware.
Sample header:
`X-Rh-Identity:ewogICJpZGVudGl0eSI6IHsKICAgICJhY2NvdW50X251bWJlciI6ICIwMzY5MjMzIiwKICAgICJ0eXBlIjogIlVzZXIiLAogICAgImF1dGhfdHlwZSI6ICJiYXNpYy1hdXRoIiwKICAgICJ1c2VyIjogewogICAgICAidXNlcm5hbWUiOiAiamRvZSIsCiAgICAgICJlbWFpbCI6ICJqZG9lQGFjbWUuY29tIiwKICAgICAgImZpcnN0X25hbWUiOiAiSm9obiIsCiAgICAgICJsYXN0X25hbWUiOiAiRG9lIiwKICAgICAgImlzX2FjdGl2ZSI6IHRydWUsCiAgICAgICJpc19vcmdfYWRtaW4iOiBmYWxzZSwKICAgICAgImlzX2ludGVybmFsIjogZmFsc2UsCiAgICAgICJsb2NhbGUiOiAiZW5fVVMiCiAgICB9LAogICAgImludGVybmFsIjogewogICAgICAib3JnX2lkIjogIjMzNDA4NTEiLAogICAgICAiYXV0aF90aW1lIjogNjMwMAogICAgfQogIH0sCiAgImVudGl0bGVtZW50cyI6IHsKICAgICJpbnNpZ2h0cyI6IHsKICAgICAgImlzX2VudGl0bGVkIjogdHJ1ZSwKICAgICAgImlzX3RyaWFsIjogZmFsc2UKICAgIH0KICB9Cn0K`

### Testing POST request:
The request should be built like this:
```
{
    "Name": "1234-auth-realm", // Uniqueness will be enforced in the DB
    "CustomResource": {},   // CustomResource object
    "Account": "0000000".   // Optional
}
```

Tests:
BadRequest if either Name/CustomResource are missing
BadRequest if account in the req body does not match the account from the request header
500 error if Name is duplicated (the exact error message text varies based on the underlying DB)

### CODE REVIEW

File: pkg/db/db.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		fmt.Println("Opening sqlite DB...")
+	}
+
+	DB, err = gorm.Open(dia, &gorm.Config{})
+	if err != nil {
+		panic(fmt.Sprintf("failed to connect database: %s", err.Error()))
+	}
+
+	// Migrate the schema
+	DB.AutoMigrate(&models.AuthRealm{})
```
Comment (robinbobbitt) [inline pkg/db/db.go:45]: Do you have details on why edge team has a separate migration step? Do we need to consider the same?

File: pkg/models/auth_realm.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+package models
+
+import (
+	"gorm.io/datatypes"
+	"gorm.io/gorm"
+)
+type AuthRealm struct  {
+	gorm.Model
+	Name		string	`json:"Name" gorm:"unique"`	// Enforce uniqueness on the name
```
Comment (robinbobbitt) [inline pkg/models/auth_realm.go:11]: Uniqueness on the name is just within the account... I think we need something like this https://stackoverflow.com/questions/30831316/how-to-specify-a-struct-with-a-multi-column-unique-index-for-gorm

File: pkg/services/auth_realms.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+	if (err != nil) {
+		// Bad request
+		errors.RespondWithBadRequest(err.Error(), w)
+		return			
+	}
+
+	// Create record for auth realm in the DB		
+	tx := db.DB.Create(&authRealm)
+	if tx.Error != nil {
```
Comment (robinbobbitt) [inline pkg/services/auth_realms.go:76]: Can we detect when this is failing on uniqueness and return a 409?

**Review by robinbobbitt — CHANGES_REQUESTED** [review]
Great work! Just added a few comments/questions.

File: pkg/db/db.go (hunk 1 above)
Comment (vidyanambiar) [reply pkg/db/db.go:45]: Yep, that's on my list to check with them. I added it here for now to be able to test. I'll follow up with Bianca on it. 👍

File: pkg/models/auth_realm.go (hunk 2 above)
Comment (vidyanambiar) [reply pkg/models/auth_realm.go:11]: Ah good catch! Thanks. I'll add the uniqueness within the account.

**Review by robinbobbitt — APPROVED** [review]
lgtm

**Pull request merged**